### PR TITLE
Added and configured lint staged and lint workflow

### LINKED ISSUE #301: Set up a workflow to run the linter on new pull requests
Set up a workflow to run eslint on PRs to catch any issues in case eslint wasn't run before committing.

### LINKED ISSUE #331: Install and configure lint-staged
Same issue as https://github.com/neurobagel/query-tool/issues/30: 

Eslint currently just runs on everything, but we want it to run only on the staged changes. This tool will enable that: https://github.com/okonet/lint-staged

### PR DESCRIPTION
**Pull request by rmanaem**
Closes #301 
Closes #331

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Install dependencies
+      run: yarn
+
+    - name: Run ESLint
+      run: npm run lint
```
Comment (surchs) [inline .github/workflows/lint.yaml:16]: could the mix of `npm` for running the `lint` and `yarn` to install the dependencies above become a problem?

File: .lintstagedrc.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "*.{js,vue}": ["npm run lint"]
```
Comment (surchs) [inline .lintstagedrc.json:2]: can we get it to run on our component tests as well? `*.cy.js`?

**Review by surchs — APPROVED** [review]
Thanks for taking this on @rmanaem :tada: . This looks good to go for me. Left a question and a suggestion, see if you agree.

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (rmanaem) [reply .github/workflows/lint.yaml:16]: I don't think so but I can change `yarn` to `npm install`

File: .lintstagedrc.json (hunk 2 above)
Comment (rmanaem) [reply .lintstagedrc.json:2]: I believe it does as is

**Pull request merged**